### ISSUE TITLE
Flaky test case

### ISSUE THREAD

**Issue by doodlewind** [toeverything/blocksuite#216]
Some test cases are known to be flaky, e.g. the `basic indent and unindent` test case can be flaky locally.

**Comment by thorseraq** [comment]
I have met this when `basic indent and unindent` fails as well, and would like to give more information :)

Pic below shows when the test accidentally fails, the state of `formattedJSX`, which is serialized from `window.store` is not updated according to the last `await pressShiftTab(page)` (Confusing for me so far).

The callstack is: [`basic indent and unindent`](https://github.com/toeverything/blocksuite/blob/3e3ef9f70896407285509c7d87e70295147c92d6/tests/list.spec.ts#L218) -> [`await assertStoreMachJSX` ](https://github.com/toeverything/blocksuite/blob/3e3ef9f70896407285509c7d87e70295147c92d6/tests/list.spec.ts#L268) -> [`
  expect(snapshot.trimStart(), formattedJSX).toEqual(formattedJSX);`](https://github.com/toeverything/blocksuite/blob/3e3ef9f70896407285509c7d87e70295147c92d6/tests/utils/asserts.ts#L337)

<img width="864" alt="image" src="https://user-images.githubusercontent.com/20554850/203685407-b62198ac-fe5a-49f3-ae8d-7cec326e30a1.png">

**Comment by doodlewind** [comment]
Some findings over recent development:

* Some asserters are sensitive to mouse movement, adding `await page.mouse.move(100, 100)` makes it stable.
* Some consecutive operations may be merged into unexpected undo stack items. Explicitly adding `page.captureSync()` in the application code makes it stable.
* To investigate a flaky test, we can mark one test case as `test.only`, then perform `npx playwright test --repeat-each=10`. It's also very helpful to run `pnpm test -- --debug` with `await page.pause()` added before certain asserters.

cc @lawvs @Himself65 @thorseraq

**Comment by doodlewind** [comment]
We can get reports about flaky test after #728, and we also have some examples about fixing flaky tests:

* 49049693f2cd9e1daa3220b41e40d78aa2b4cb32
* 8c809898142cffb9ca4a709239ea205ea87a75f1
* 294e5b314205b32af03c6817a5571c4c1fa911ba
* 7bf699ce6c1da73a4199eee77771a4f88f16e0cd
* 6f113db0c4c6feab935b3fa2bd54fe53036ec2ff

They are mostly about timeout.

Closing this issue since we plan to fix other flaky tests case-by-case.
